src/zenml/integrations/gcp/services/vertex_deployment.py

```diff
@@ -0,0 +1,431 @@
+        )
+
+        zenml_client = Client()
+        model_deployer = zenml_client.active_stack.model_deployer
+        if not isinstance(model_deployer, VertexModelDeployer):
+            raise RuntimeError(
+                "Active model deployer must be Vertex AI Model Deployer"
+            )
+
+        # get credentials from model deployer
+        credentials, project_id = model_deployer._get_authentication()
+
+        # Initialize aiplatform
+        aiplatform.init(
+            project=project_id,
+            location=self.config.location,
+            credentials=credentials,
+        )
+
+        # Initialize logging client
+        self.logging_client = vertex_logging.Client(
+            project=project_id, credentials=credentials
+        )
+
+    def __init__(self, config: VertexDeploymentConfig, **attrs: Any):
+        """Initialize the Vertex AI deployment service."""
+        super().__init__(config=config, **attrs)
+        self._initialize_gcp_clients()
+
+    @property
+    def prediction_url(self) -> Optional[str]:
+        """The prediction URI exposed by the prediction service."""
+        if not self.status.endpoint or not self.status.endpoint.endpoint_url:
+            return None
+
+        # Construct proper prediction URL
+        return f"https://{self.config.location}-aiplatform.googleapis.com/v1/{self.status.endpoint.endpoint_url}"
+
+    def get_endpoints(self) -> List[aiplatform.Endpoint]:
+        """Get all endpoints for the current project and location."""
+        try:
+            # Use proper filtering and pagination
+            return list(
+                aiplatform.Endpoint.list(
+                    filter='labels.managed_by="zenml"',
+                    location=self.config.location,
+                )
+            )
+        except Exception as e:
+            logger.error(f"Failed to list endpoints: {e}")
+            return []
+
+    def _generate_endpoint_name(self) -> str:
+        """Generate a unique name for the Vertex AI Inference Endpoint."""
+        # Make name more descriptive and conformant to Vertex AI naming rules
+        sanitized_model_name = re.sub(
+            r"[^a-zA-Z0-9-]", "-", self.config.model_name.lower()
+        )
+        return f"{sanitized_model_name}-{str(self.uuid)[:UUID_SLICE_LENGTH]}"
+
+    def provision(self) -> None:
+        """Provision or update remote Vertex AI deployment instance."""
+        try:
+            if self.config.existing_endpoint:
+                # Use the existing endpoint
+                endpoint = aiplatform.Endpoint(
+                    endpoint_name=self.config.existing_endpoint,
+                    location=self.config.location,
+                )
+                logger.info(
+                    f"Using existing Vertex AI inference endpoint: {endpoint.resource_name}"
+                )
+            else:
+                # Create the endpoint
+                endpoint_name = self._generate_endpoint_name()
```

Comment:
same thing here: multiple zenml tenants or servers might want to use the same model name, and therefore might clash on the same endpoint name. You could avoid this by using the server UUID to differentiate/filter endpoint resources and also as part of the endpoint name.